File: lib/theme.ts
import { Platform, TextStyle } from 'react-native';

export const colors = {
  ink: '#0A0A0A',
  black: '#000000',
  white: '#FFFFFF',
  mute: '#6B7280',
  mute2: '#9CA3AF',
  border: '#E5E7EB',
  surface: '#F4F4F5',
  surface2: '#FAFAFA',
  success: '#10B981',
  successSoft: '#D1FAE5',
  danger: '#EF4444',
  dangerSoft: '#FEE2E2',
  warn: '#F59E0B',
  warnSoft: '#FEF3C7',
  info: '#3B82F6',
  infoSoft: '#DBEAFE',
} as const;

export const radius = {
  xs: 8,
  sm: 12,
  md: 16,
  lg: 20,
  xl: 28,
  pill: 999,
} as const;

export const space = {
  xs: 4,
  s: 8,
  m: 12,
  l: 16,
  xl: 24,
  xxl: 32,
  xxxl: 48,
} as const;

const sfFamily = Platform.select({
  ios: 'System',
  android: 'sans-serif',
  default: '-apple-system, BlinkMacSystemFont, "SF Pro Display", "Inter", system-ui, sans-serif',
});

export const type = {
  display: {
    fontFamily: sfFamily,
    fontSize: 44,
    lineHeight: 48,
    fontWeight: '800',
    letterSpacing: -1,
  } satisfies TextStyle,
  h1: {
    fontFamily: sfFamily,
    fontSize: 32,
    lineHeight: 36,
    fontWeight: '800',
    letterSpacing: -0.6,
  } satisfies TextStyle,
  h2: {
    fontFamily: sfFamily,
    fontSize: 24,
    lineHeight: 28,
    fontWeight: '700',
    letterSpacing: -0.3,
  } satisfies TextStyle,
  h3: {
    fontFamily: sfFamily,
    fontSize: 18,
    lineHeight: 22,
    fontWeight: '700',
    letterSpacing: -0.2,
  } satisfies TextStyle,
  body: {
    fontFamily: sfFamily,
    fontSize: 16,
    lineHeight: 24,
    fontWeight: '400',
  } satisfies TextStyle,
  bodyBold: {
    fontFamily: sfFamily,
    fontSize: 16,
    lineHeight: 24,
    fontWeight: '600',
  } satisfies TextStyle,
  small: {
    fontFamily: sfFamily,
    fontSize: 13,
    lineHeight: 18,
    fontWeight: '400',
  } satisfies TextStyle,
  smallBold: {
    fontFamily: sfFamily,
    fontSize: 13,
    lineHeight: 18,
    fontWeight: '600',
  } satisfies TextStyle,
  micro: {
    fontFamily: sfFamily,
    fontSize: 11,
    lineHeight: 14,
    fontWeight: '700',
    letterSpacing: 0.4,
    textTransform: 'uppercase' as const,
  } satisfies TextStyle,
} as const;

export const shadow = {
  card: Platform.select({
    ios: {
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 2 },
      shadowOpacity: 0.05,
      shadowRadius: 8,
    },
    android: { elevation: 2 },
    default: { boxShadow: '0 2px 8px rgba(0,0,0,0.05)' as unknown as never },
  }),
  lift: Platform.select({
    ios: {
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 8 },
      shadowOpacity: 0.12,
      shadowRadius: 24,
    },
    android: { elevation: 8 },
    default: { boxShadow: '0 8px 24px rgba(0,0,0,0.12)' as unknown as never },
  }),
} as const;

export function fmtUSD(amount: number, options: Intl.NumberFormatOptions = {}): string {
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
    ...options,
  }).format(amount);
}

File: ios-native/SubSpy/Theme/Theme.swift
import SwiftUI

enum Palette {
    static let ink         = Color(red: 0.039, green: 0.039, blue: 0.039)
    static let black       = Color.black
    static let white       = Color.white
    static let mute        = Color(red: 0.42, green: 0.45, blue: 0.50)
    static let mute2       = Color(red: 0.61, green: 0.64, blue: 0.69)
    static let border      = Color(red: 0.898, green: 0.906, blue: 0.922)
    static let surface     = Color(red: 0.957, green: 0.957, blue: 0.961)
    static let surface2    = Color(red: 0.98, green: 0.98, blue: 0.98)
    static let success     = Color(red: 0.063, green: 0.725, blue: 0.506)
    static let successSoft = Color(red: 0.82, green: 0.98, blue: 0.91)
    static let danger      = Color(red: 0.937, green: 0.267, blue: 0.267)
    static let dangerSoft  = Color(red: 0.996, green: 0.894, blue: 0.886)
    static let warn        = Color(red: 0.961, green: 0.620, blue: 0.043)
    static let warnSoft    = Color(red: 0.996, green: 0.953, blue: 0.776)
    static let info        = Color(red: 0.231, green: 0.510, blue: 0.965)
    static let infoSoft    = Color(red: 0.859, green: 0.918, blue: 0.996)
    static let zombieFg    = Color(red: 0.6, green: 0.106, blue: 0.110)
    static let reviewFg    = Color(red: 0.573, green: 0.251, blue: 0.055)
    static let keepFg      = Color(red: 0.024, green: 0.373, blue: 0.275)
    static let infoFg      = Color(red: 0.118, green: 0.251, blue: 0.686)
}

enum Radius {
    static let xs: CGFloat = 8
    static let sm: CGFloat = 12
    static let md: CGFloat = 16
    static let lg: CGFloat = 20
    static let xl: CGFloat = 28
    static let pill: CGFloat = 999
}

enum Space {
    static let xs: CGFloat = 4
    static let s: CGFloat = 8
    static let m: CGFloat = 12
    static let l: CGFloat = 16
    static let xl: CGFloat = 24
    static let xxl: CGFloat = 32
    static let xxxl: CGFloat = 48
}

enum AppFont {
    static let display = Font.system(size: 44, weight: .heavy)
    static let h1      = Font.system(size: 32, weight: .heavy)
    static let h2      = Font.system(size: 24, weight: .bold)
    static let h3      = Font.system(size: 18, weight: .bold)
    static let body    = Font.system(size: 16, weight: .regular)
    static let bodyB   = Font.system(size: 16, weight: .semibold)
    static let small   = Font.system(size: 13, weight: .regular)
    static let smallB  = Font.system(size: 13, weight: .semibold)
    static let micro   = Font.system(size: 11, weight: .bold)
}

extension Text {
    func micro() -> some View {
        self.font(AppFont.micro)
            .tracking(0.4)
            .textCase(.uppercase)
    }
}

func fmtUSD(_ amount: Double) -> String {
    let f = NumberFormatter()
    f.numberStyle = .currency
    f.currencyCode = "USD"
    f.locale = Locale(identifier: "en_US")
    f.minimumFractionDigits = 2
    f.maximumFractionDigits = 2
    return f.string(from: NSNumber(value: amount)) ?? "$\(amount)"
}

func fmtRelDate(_ date: Date) -> String {
    let f = DateFormatter()
    f.locale = Locale(identifier: "en_US")
    f.dateFormat = "MMM d, yyyy"
    return f.string(from: date)
}
